src/wp-includes/html-api/class-wp-html-spec.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+class WP_HTML_Spec {
```

Comment:
❤️

Reply:
this is something I've meant to create in the HTML processor branches I've been working in. the idea is to move utilities which related to HTML-defined terminology into here.

that would be:
 - `is_void_element()`
 - `is_html_element()`
 - `is_format_element()`
 - etc…

`is_html_element()` will be important for HTML parsing because _foreign elements_ may be self-closing, but HTML elements may not be, and therefore we have to ignore the trailing `/` in something like an `<img />` tag and look for a closing tag if it's not a void element, such as with `<div />` (because while it looks like a self-closer it _isn't_), but `<wp-self-closer />` _is_ due to it being a foreign element.